models/seir/model_mod.f90

```diff
@@ -0,0 +1,455 @@
+end do
+
+! This time is both the minimum time you can ask the model to advance
+! (for models that can be advanced by filter) and it sets the assimilation
+! window.  All observations within +/- 1/2 this interval from the current
+! model time will be assimilated. If this isn't settable at runtime 
+! feel free to hardcode it and not add it to a namelist.
+time_step = set_time(time_step_seconds, time_step_days)
+
+! Tell the DART I/O routines how large the model data is so they
+! can read/write it.
+dom_id = add_domain(model_size)
+
+end subroutine static_init_model
+
+!------------------------------------------------------------------
+! Advance the SEIR model using a four-step RK scheme
+
+subroutine adv_1step(x, time)
+
+real(r8), intent(inout)      :: x(:)
+type(time_type), intent(in)  :: time
+
+real(r8), dimension(size(x)) :: xi, x1, x2, x3, x4, dx 
+
+! 1st step
+call seir_eqns(x, dx)   
+x1 = delta_t * dx
+
+! 2nd step
+xi = x + 0.5_r8 * delta_t * dx
+call seir_eqns(xi, dx)  
+x2 = delta_t * dx
+
+! 3rd step
+xi = x + 0.5_r8 * delta_t * dx
+call seir_eqns(xi, dx)    
+x3 = delta_t * dx
+
+! 4th step
+xi = x + delta_t * dx
+call seir_eqns(xi, dx)  
+x4 = delta_t * dx
+
+! Compute new value for x
+x = x + x1/6.0_r8 + x2/3.0_r8 + x3/3.0_r8 + x4/6.0_r8
+
+end subroutine adv_1step
+
+!------------------------------------------------------------------
+! SEIR Model Equations
+! The following extended SEIR Model with Vaccination 
+! is adaopted from Ghostine et al. (2021):
+! Ghostine, R., Gharamti, M., Hassrouny, S and Hoteit, I 
+! "An Extended SEIR Model with Vaccination for Forecasting
+! the COVID-19 Pandemic in Saudi Arabia Using an Ensemble
+! Kalman Filter" Mathematics 2021, 9, 636.
+! https://dx.doi.org/10.3390/math9060636 
+
+subroutine seir_eqns(x, fx)
+
+! State: x = [S, E, I, Q, R, D, V]
+
+real(r8), intent(in)  :: x(:)
+real(r8), intent(out) :: fx(:)
+
+fx(1) = theta - beta * x(1) * x(3) - alpha * x(1) - mu * x(1)
+fx(2) = beta * x(1) * x(3) - gama * x(2) + sigma * beta * x(7) * x(3) - mu * x(2)
+fx(3) = gama * x(2) - delta * x(3) - mu * x(3)
+fx(4) = delta * x(3) - (1.0_r8 - kappa) * lambda * x(4) - kappa * rho * x(4) - mu * x(4)
+fx(5) = (1.0_r8 - kappa) * lambda * x(4) - mu * x(5)
+fx(6) = kappa * rho * x(4)
+fx(7) = alpha * x(1) - sigma * beta * x(7) * x(3) - mu * x(7)
+
+end subroutine seir_eqns
+
+!------------------------------------------------------------------
+! Perturbs a model state for generating initial ensembles.
+! Returning interf_provided .true. means this code has
+! added uniform small independent perturbations to a
+! single ensemble member to generate the full ensemble.
+subroutine pert_model_copies(state_ens_handle, ens_size, pert_amp, interf_provided)
+
+type(ensemble_type), intent(inout) :: state_ens_handle
+integer,   intent(in) :: ens_size
+real(r8),  intent(in) :: pert_amp
+logical,  intent(out) :: interf_provided
+
+integer                  :: i,j, num_my_grid_points
```

Comment:
`i` should be `i8` since you might have the whole state (billions of people) on one task.